### PR TITLE
.Net Processes: Fixing an issue with nested processes in Dapr runtime.

### PR DESCRIPTION
**Pull request by alliscode**
### Description

This PR addresses an issue when running a nested process in the Dapr runtime. After this changes, the Dapr runtime fully supports nested processes with the ability to pass events in both directions.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Experimental/Process.LocalRuntime/LocalProcess.cs [hunk 1]
```diff
@@ -188,7 +188,6 @@ private ValueTask InitializeProcessAsync()
                     kernel: this._kernel,
                     parentProcessId: this.Id);
 
-                //await process.StartAsync(kernel: this._kernel, keepAlive: true).ConfigureAwait(false);
```
Comment (crickman) [inline dotnet/src/Experimental/Process.LocalRuntime/LocalProcess.cs:191]: Heh, I removed this in the global-error-handler banch also! :smile:

File: dotnet/src/Experimental/Process.Runtime.Dapr/DaprStepInfo.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ namespace Microsoft.SemanticKernel;
 /// </summary>
 [KnownType(typeof(KernelProcessEdge))]
 [KnownType(typeof(KernelProcessStepState))]
+[KnownType(typeof(DaprProcessInfo))]
```
Comment (crickman) [inline dotnet/src/Experimental/Process.Runtime.Dapr/DaprStepInfo.cs:15]: Ok, this is where I was also placing this...will add `DaprMapInfo` here also. 👍

**Review by crickman — APPROVED** [review]

File: dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs [hunk 3]
```diff
@@ -383,10 +412,28 @@ private async Task<DaprProcessInfo> ToDaprProcessInfoAsync()
...
+    /// <summary>
+    /// Generates a scoped event for the step.
+    /// </summary>
+    /// <param name="daprEvent">The event.</param>
+    /// <returns>A <see cref="ProcessEvent"/> with the correctly scoped namespace.</returns>
+    private ProcessEvent ScopedEvent(ProcessEvent daprEvent)
     {
-        return new ActorId($"{this.Id}.{actorId.GetId()}");
+        Verify.NotNull(daprEvent);
+        return daprEvent with { Namespace = $"{this.Name}_{this._process!.State.Id}" };
```
Comment (esttenorio) [inline dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs:436]: should there be common `namespace name generator` to ensure the namespace name remains consistent? 
I see this string format is repeated in other files too

**Review by esttenorio — APPROVED** [review]

File: dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs [hunk 4]
```diff
@@ -308,8 +308,7 @@ private async Task Internal_ExecuteAsync(Kernel? kernel = null, int maxSuperstep
                 await Task.WhenAll(stepProcessingTasks).ConfigureAwait(false);
 
                 // Handle public events that need to be bubbled out of the process.
-                var eventQueue = this.ProxyFactory.CreateActorProxy<IEventBuffer>(new ActorId(this.Id.GetId()), nameof(EventBufferActor));
```
Comment (moonbox3) [inline dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs:311]: Can we remove `localKernel` and `messageChannel` on lines 274/275? I don't see them being used.

**Review by moonbox3 — APPROVED** [review]

File: dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs (hunk 3 above)
Comment (alliscode) [reply dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs:436]: This one is actually different, the scoping is a bit different between the step and the (nested) process. But I have it on my backlog to refactor the scoping of things to be clearer and less error prone.

File: dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs (hunk 4 above)
Comment (alliscode) [reply dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs:311]: Yes, good catch

**Review by moonbox3 — APPROVED** [review]

**Review by crickman — APPROVED** [review]

**Pull request merged**